ide/extexecution.base/src/org/netbeans/api/extexecution/base/BaseExecutionService.java

```diff
@@ -140,6 +140,43 @@ public static BaseExecutionService newService(@NonNull Callable<? extends Proces
         return new BaseExecutionService(processCreator, descriptor);
     }
 
+    /**
+     * Infers the output encoding from the relevant system properties, if those should all be <code>null</code>
+     * then this will fallback to <code>Charset.defaultCharset()</code>
+     * 
+     * Since JDK 18 and JEP400 Console.charset() is used for the console's encoding instead of <code>Charset.defaultCharset()</code>. 
+     * Console.charset() is exposed via stdout.encoding/sun.stdout.encoding.
+     * If ran with JDK<=16 these properties should be null and the old default of <code>Charset.defaultCharset()</code> will be used to match pre JEP400 behavior.
+     * 
+     * The checking order for the encoding is stdout.encoding, sun.stdout.encoding, native.encoding, <code>Charset.defaultCharset()</code>
+     * 
+     * @return inferred encoding as Charset
+     */
+    private static Charset getInputOutputEncoding(){
+        String[] encodingSystemProperties = new String[]{"stdout.encoding, sun.stdout.encoding, native.encoding"};
+
+        Charset preferredCharset = null;
+        for(String encodingProperty : encodingSystemProperties){
+            String encodingPropertyValue = System.getProperty(encodingProperty);
+            if(encodingPropertyValue == null){
+                continue;
+            }
+
+            try {
+                preferredCharset = Charset.forName(encodingPropertyValue);
+            } catch (Exception ex) {
+                LOGGER.log(java.util.logging.Level.WARNING, "Failed to get charset for '" + encodingProperty + "' value : '" + encodingPropertyValue + "'", ex);
+            }
+
+            if(preferredCharset != null){
```

Comment:
Please correct the spacing.

Reply:
![spacing](https://user-images.githubusercontent.com/50800372/179525254-36d576da-c4a7-42ef-beec-f04da910a15f.PNG)

Is all of 167-171 spaced wrongly? Because line 168 should have the same number of spaces line 172 (16 spaces)

Reply:
eg. that line should be `if (preferredCharset != null) {` - ie. please add spaces around brackets and braces.

The easiest way is to select the sections you've changed (_and only the sections you've changed!_) and auto format it (`Alt+Shift+F`)

Reply:
btw I just noticed that you only fixed one occurrence, The spaces are also missing at the for loop, the other if and one of the two method declarations.

Reply:
Hopefully I got every spacing error now; I noticed that I had set my IDE to the wrong conventions. Thus Autoformat caused these.